tensorflow/compiler/tf2tensorrt/utils/trt_tensor_proxy.h

```diff
@@ -0,0 +1,489 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_COMPILER_TF2TENSORRT_CONVERT_TRT_TENSOR_PROXY_H
+#define TENSORFLOW_COMPILER_TF2TENSORRT_CONVERT_TRT_TENSOR_PROXY_H
+
+#include <vector>
+#include <string>
+#include <cassert>
+#include <memory>
+#include "tensorflow/compiler/tf2tensorrt/common/utils.h"
+
+#if GOOGLE_CUDA && GOOGLE_TENSORRT
+#include "third_party/tensorrt/NvInfer.h"
+
+namespace tensorflow {
+
+namespace tensorrt {
+
+namespace {
+  nvinfer1::Dims GetTestDims(const std::vector<int>& d) {
+    nvinfer1::Dims dims;
+    dims.nbDims = d.size();
+    for (int i = 0; i < d.size(); ++i) {
+      dims.d[i] = d[i];
+    }
+  }
+}
```

Comment:
Using unnamed namespace in header can cause problem (see [here](https://rules.sonarsource.com/cpp/RSPEC-1000)) and is prohibited by c++ coding style ([here](https://google.github.io/styleguide/cppguide.html#Internal_Linkage)).